WINGSTOP_Air_Brake_Flight_Computer_FULL_BRAKES/WINGSTOP_Air_Brake_Flight_Computer_FULL_BRAKES.ino

```diff
@@ -195,128 +228,138 @@ void setup() {
-  x1 = read_altitude();
-  //*///END SECTION 1 
-}
-  /* SECTION 2
-  // UNCOMMENT this section to use all sea level data. Target altitude will be updated using initial altitude reading and target above ground. Must comment first section. 
-  init_pressure = SEALEVELPRESSURE_HPA;
-  init_altitude = read_altitude();
-  x1 = init_altitude;
-  target_altitude += init_altitude;
-  *///END SECTION 2 
+  delay(1000); //take altitude reading after one second
+  x_current = bmp.readAltitude(init_pressure);
+  
+  set_header_file(); //sets headers at beginning of file
+
+  if (abs(x_current - x_previous) > 3 ){//enters if roket is moving greater than 3 m/s
+    logFile.print(F("ABORTED, ROCKET MOVING"));
+    beep_buzz(1); //beeps and lights once to signal that rocket was detected moving
+    while (1){//runs forever
+      //update altitude at frequency (hz)
+      average_acceleration_data(); // updates acceleration value acc, and delta_t value
+      read_velocity();//reads new altitude as x_current and compares to previous altitude x_previous and delta_t to determine velocity, then updates x_previous as x_current
+      log_data();//logs data to sd card
+      reopen_file();//saves and reopens file
+      
+    }
+  }
 
   
-  //print initial sea level altitude to file
-  myFile.print(init_altitude);
 
-  //sets initial time, t1
-  t1 = millis();
+  //sets initial time, t_previous
+  t_previous = millis();
   
   //LED light On Solid to show ARMED. Turns off when motor ignition detected. 
-  turn_on_led(); //Turn on LED
+  beep_buzz(3); //beeps and blinks LED 3 times to indicate armed
 
+  
   //COMPUTER IS NOW ARMED
 }
 
 
 
 
 
+
+
+
 /////////MAIN PROGRAM///////////////MAIN PROGRAM//////////////MAIN PROGRAM///////////////MAIN PROGRAM////////////////MAIN PROGRAM////////////////MAIN PROGRAM///////////////MAIN PROGRAM/////////
 //BEGIN LOOP MAIN PROGRAM
 
 void loop() { 
 
 
 //STANDBY MODE********************
-  while(vel < 4){//Waiting for launch. Exits if velocity is greater than 4 m/s.
+  while(!detect_take_off()){//Waiting for launch. Exits if altitude is above takeoff altitude
     average_acceleration_data(); // updates acceleration value acc, and delta_t value
-    read_velocity();//reads new altitude as x2 and compares to previous altitude x1 and delta_t to determine velocity, the updates x1 as x2
-    if(c % (5*hz) == 0){//turns on buzzer every 5 seconds
-      turn_on_buzzer();
-    }
-    if((c - (hz/4)) % (5*hz) == 0){//turns off buzzer after 1/4 of a second
-      turn_off_buzzer();
-    }
-    c++;//increment counter for buzzer
+    read_velocity();//reads new altitude as x_current and compares to previous altitude x_previous and delta_t to determine velocity, the updates x_previous as x_current
   }
 
-  c = 0; //reset counter for MCO detection
-  turn_off_led();
-  turn_off_buzzer();
   log_data();//logs all data to sd card
-  myFile.print(F("Liftoff Detected"));
 
 
+  if (logFile) {
```

Comment:
This block can go in a function, similar to [this](https://github.com/Palouse-Aerospace-2023-Spaceport-Cup/SRAD-Deployment-Computer/pull/2#discussion_r1096883486)